Deferred/Deferred.swift

```diff
@@ -135,7 +205,15 @@ extension Deferred {
         }
         return d
     }
-
+    
+    /**
+    Transforms the deferred value once it becomes determined.
```

Comment:
I think a `:note:` here and on `flatMap` to point out the distinction between them and why you'd use one vs the other might be very helpful. The param and returns docs are identical right now, which further confuses the issue.